### DISCUSSION TITLE
search is not a function - React, jQuery

### DISCUSSION THREAD

**Discussion by ghost** [fomantic/Fomantic-UI#1591] [General] [answered]
# Bug Report
Using ui search with local static source and jquery throwing error that search is not a function.

--> https://fomantic-ui.com/modules/search.html 
--> https://fomantic-ui.com/modules/search.html#local-search

## Steps to reproduce
1) React contains return with
`            <div className="ui search">  
                <div class="ui icon input">  
                  <input class="prompt" type="text" placeholder="Search countries..." />  
                  <i class="search icon"></i>  
                </div>  
                <div class="results"></div>  
            </div>`  

2) componentDidMount contains

`         $('.ui.search').search({
            source: content
          })
        ;`

3) jQuery is installed

## Expected result
Returns input that enables search within the local source. In this case within anything that is in variable named source which is a javascript ojbect.

## Actual result
TypeError: jquery__WEBPACK_IMPORTED_MODULE_9___default(...)(...).search is not a function

## Version

jquery: "^3.5.1"
react: "^16.13.1"
react-dom: "^16.13.1

**Comment by lubber-de** [comment]
Is jquery included _before_ fomantic js included? Please prepare codepen to reproduce your issue

**Comment by lubber-de** [comment]
I believe your webpack configuration imports in the wrong order. jquery has to be imported before fomantic.js is.
Maybe this issue helps https://github.com/webpack/webpack/issues/1343

**Comment by ko2in** [comment]
Maybe, this could also help (https://webpack.js.org/plugins/provide-plugin/)

**Comment by ghost** [comment]
Thank you guys for your answers.

When I locally do this : https://jsfiddle.net/xchochu/vpkrmocf/16/  (I also locally import react and react-dom), the app throws me an error **'$' is not defined**. When I import $ from jquery, then I recieve an error I already mentioned above. 

I also uploaded the code snippet to the github, which can be found here - https://github.com/pchochu/someApp

**Comment by lubber-de** [comment, marked as answer]
@pchochu Thanks for the code snippet. I could reproduce (and solve 🙂 ) it.

First. **this is a general webpack issue for all jquery plugins**, The final hint for the easiest fix is here https://github.com/webpack/webpack/issues/4258#issuecomment-340240162

#### To fix this in your given code
- Remove all your cdn links for fomantic or jquery in `index.html`. They are useless/overwritten by webpack bundle (and you even included jquery twice)
- install jquery and fomantic-ui-css via npm

```console
npm add jquery
npm add fomantic-ui-css
```

Then, in your `index.js` change the top code to the following to make it work

```javascript
import React from 'react'
import ReactDOM from 'react-dom'
import 'fomantic-ui-css/semantic.css';
import $ from 'jquery'
window.jQuery = $;   //this makes jquery accessible for any later jquery plugin (like FUI in this case)
window.$ = $;
require('fomantic-ui-css/semantic.js'); // import does not work here anymore

```

if you don't like this approach (webpack has more "[standardized](https://webpack.js.org/configuration/externals/#externals)" ways which, however need much more adjustments) and still want to use webpack , please refer to the webpack docs and study the above mentioned issue. As said it's a known jquery plugins problem at webpack.

Closing, because not FUI dependent and solution is given
